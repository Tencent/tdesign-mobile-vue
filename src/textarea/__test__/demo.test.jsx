/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import autosizeVue from '@/textarea/demos/autosize.vue';
import baseVue from '@/textarea/demos/base.vue';
import cardVue from '@/textarea/demos/card.vue';
import customVue from '@/textarea/demos/custom.vue';
import disabledVue from '@/textarea/demos/disabled.vue';
import labelVue from '@/textarea/demos/label.vue';
import maxcharacterVue from '@/textarea/demos/maxcharacter.vue';
import maxlengthVue from '@/textarea/demos/maxlength.vue';
import mobileVue from '@/textarea/demos/mobile.vue';

const mapper = {
  autosizeVue,
  baseVue,
  cardVue,
  customVue,
  disabledVue,
  labelVue,
  maxcharacterVue,
  maxlengthVue,
  mobileVue,
};

describe('Textarea', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Textarea ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
