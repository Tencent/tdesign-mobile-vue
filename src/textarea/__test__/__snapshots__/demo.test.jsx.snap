// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Textarea > Textarea autosizeVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal"
>
  <!---->
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner"
      name="标签文字"
      placeholder="请输入文字"
    />
    <!---->
  </div>
</div>
`;

exports[`Textarea > Textarea baseVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal textarea-example"
  data-v-dc028088=""
>
  <!---->
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner"
      name=""
      placeholder="请输入文字"
    />
    <!---->
  </div>
</div>
`;

exports[`Textarea > Textarea cardVue demo works fine 1`] = `
<div
  data-v-app=""
>
  
  <div
    class="textarea-example"
    data-v-0fc6070a=""
  >
    <div
      class="t-textarea t-textarea--layout-vertical textarea"
      data-v-0fc6070a=""
    >
      <div
        class="t-textarea__label"
      >
        标签文字
      </div>
      <div
        class="t-textarea__wrapper"
      >
        <textarea
          class="t-textarea__wrapper-inner"
          maxlength="500"
          name=""
          placeholder="请输入文字"
        />
        <div
          class="t-textarea__indicator"
        >
          0/500
        </div>
      </div>
    </div>
  </div>
  <div
    class="textarea-example card"
    data-v-0fc6070a=""
  >
    <div
      class="textarea-example__summary"
      data-v-0fc6070a=""
    >
      卡片样式
    </div>
    <div
      class="t-textarea t-textarea--layout-horizontal textarea"
      data-v-0fc6070a=""
    >
      <div
        class="t-textarea__label"
      >
        标签文字
      </div>
      <div
        class="t-textarea__wrapper"
      >
        <textarea
          class="t-textarea__wrapper-inner"
          maxlength="500"
          name=""
          placeholder="请输入文字"
        />
        <div
          class="t-textarea__indicator"
        >
          0/500
        </div>
      </div>
    </div>
  </div>
  
</div>
`;

exports[`Textarea > Textarea customVue demo works fine 1`] = `
<div
  class="textarea-example"
  data-v-1eb98b9c=""
>
  <text
    class="textarea-example__label"
    data-v-1eb98b9c=""
  >
    标签文字
  </text>
  <div
    class="t-textarea t-textarea--layout-horizontal t-textarea--border textarea"
    data-v-1eb98b9c=""
  >
    <!---->
    <div
      class="t-textarea__wrapper"
    >
      <textarea
        class="t-textarea__wrapper-inner"
        maxlength="100"
        name=""
        placeholder="请输入文字"
      />
      <div
        class="t-textarea__indicator"
      >
        0/100
      </div>
    </div>
  </div>
</div>
`;

exports[`Textarea > Textarea disabledVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal textarea-example"
  data-v-43d18467=""
>
  <div
    class="t-textarea__label"
  >
    标签文字
  </div>
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner t-textarea--disabled"
      disabled=""
      name=""
      placeholder="请输入文字"
    />
    <!---->
  </div>
</div>
`;

exports[`Textarea > Textarea labelVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal textarea-example"
  data-v-0ef6c9ce=""
>
  <div
    class="t-textarea__label"
  >
    标签文字
  </div>
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner"
      name="标签文字"
      placeholder="请输入文字"
    />
    <!---->
  </div>
</div>
`;

exports[`Textarea > Textarea maxcharacterVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal"
>
  <div
    class="t-textarea__label"
  >
    标签文字
  </div>
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner"
      name="标签文字"
      placeholder="请输入文字"
    />
    <div
      class="t-textarea__indicator"
    >
      0/500
    </div>
  </div>
</div>
`;

exports[`Textarea > Textarea maxlengthVue demo works fine 1`] = `
<div
  class="t-textarea t-textarea--layout-horizontal"
>
  <div
    class="t-textarea__label"
  >
    标签文字
  </div>
  <div
    class="t-textarea__wrapper"
  >
    <textarea
      class="t-textarea__wrapper-inner"
      maxlength="500"
      name="标签文字"
      placeholder="请输入文字"
    />
    <div
      class="t-textarea__indicator"
    >
      0/500
    </div>
  </div>
</div>
`;

exports[`Textarea > Textarea mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Textarea 多行文本框
  </h1>
  <p
    class="summary"
  >
    用于多行文本的输入
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        01 组件类型
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        基础多行文本框
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal textarea-example"
        data-v-dc028088=""
      >
        <!---->
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner"
            name=""
            placeholder="请输入文字"
          />
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        带标题多行文本框
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal textarea-example"
        data-v-0ef6c9ce=""
      >
        <div
          class="t-textarea__label"
        >
          标签文字
        </div>
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner"
            name="标签文字"
            placeholder="请输入文字"
          />
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        自动增高多行文本框
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal"
      >
        <!---->
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner"
            name="标签文字"
            placeholder="请输入文字"
          />
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <!--v-if-->
      <p
        class="tdesign-mobile-demo-block__summary tdesign-mobile-demo-block_subtitle"
      >
        设置字符数限制
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal"
      >
        <div
          class="t-textarea__label"
        >
          标签文字
        </div>
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner"
            maxlength="500"
            name="标签文字"
            placeholder="请输入文字"
          />
          <div
            class="t-textarea__indicator"
          >
            0/500
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block tdesign-mobile-demo-block_notitle"
  >
    <!--v-if-->
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal"
      >
        <div
          class="t-textarea__label"
        >
          标签文字
        </div>
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner"
            name="标签文字"
            placeholder="请输入文字"
          />
          <div
            class="t-textarea__indicator"
          >
            0/500
          </div>
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        02 组件状态
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        禁用状态
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="t-textarea t-textarea--layout-horizontal textarea-example"
        data-v-43d18467=""
      >
        <div
          class="t-textarea__label"
        >
          标签文字
        </div>
        <div
          class="t-textarea__wrapper"
        >
          <textarea
            class="t-textarea__wrapper-inner t-textarea--disabled"
            disabled=""
            name=""
            placeholder="请输入文字"
          />
          <!---->
        </div>
      </div>
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        03 组件样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        竖排样式
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      
      <div
        class="textarea-example"
        data-v-0fc6070a=""
      >
        <div
          class="t-textarea t-textarea--layout-vertical textarea"
          data-v-0fc6070a=""
        >
          <div
            class="t-textarea__label"
          >
            标签文字
          </div>
          <div
            class="t-textarea__wrapper"
          >
            <textarea
              class="t-textarea__wrapper-inner"
              maxlength="500"
              name=""
              placeholder="请输入文字"
            />
            <div
              class="t-textarea__indicator"
            >
              0/500
            </div>
          </div>
        </div>
      </div>
      <div
        class="textarea-example card"
        data-v-0fc6070a=""
      >
        <div
          class="textarea-example__summary"
          data-v-0fc6070a=""
        >
          卡片样式
        </div>
        <div
          class="t-textarea t-textarea--layout-horizontal textarea"
          data-v-0fc6070a=""
        >
          <div
            class="t-textarea__label"
          >
            标签文字
          </div>
          <div
            class="t-textarea__wrapper"
          >
            <textarea
              class="t-textarea__wrapper-inner"
              maxlength="500"
              name=""
              placeholder="请输入文字"
            />
            <div
              class="t-textarea__indicator"
            >
              0/500
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <div
      class="tdesign-mobile-demo-block__header"
    >
      <h2
        class="tdesign-mobile-demo-block__title"
      >
        04 特殊样式
      </h2>
      <p
        class="tdesign-mobile-demo-block__summary"
      >
        标签外置输入框
      </p>
    </div>
    <div
      class="tdesign-mobile-demo-block__slot"
    >
      
      <div
        class="textarea-example"
        data-v-1eb98b9c=""
      >
        <text
          class="textarea-example__label"
          data-v-1eb98b9c=""
        >
          标签文字
        </text>
        <div
          class="t-textarea t-textarea--layout-horizontal t-textarea--border textarea"
          data-v-1eb98b9c=""
        >
          <!---->
          <div
            class="t-textarea__wrapper"
          >
            <textarea
              class="t-textarea__wrapper-inner"
              maxlength="100"
              name=""
              placeholder="请输入文字"
            />
            <div
              class="t-textarea__indicator"
            >
              0/100
            </div>
          </div>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
